updater/internal/state/monitor.go

```diff
@@ -111,11 +110,11 @@ func (c *CollectorMonitor) MonitorForSuccess(ctx context.Context, packageName st
 		case <-ticker.C:
 			packageStatus, err := c.stateManager.LoadStatuses()
 			switch {
-			case errors.Is(err, os.ErrNotExist):
-				// File may not exist for a short period of time while being written so just continue
-				continue
+			// If there is any error we'll just continue. Some valid reasons we could error and should retry:
+			// - File was deleted by new collector before it's rewritten
+			// - File is being written to while we're reading it so we'll get invalid JSON
 			case err != nil:
-				return fmt.Errorf("error while loaded status: %w", err)
+				continue
```

Comment:
should we still log the error?

Reply:
There currently isn't a logger in this thing. Probably should add it though.

Reply:
Actually, the underlying state manager logs this error so we won't need to. That's why I didn't add a logger.